vcd/resource_vcd_dnat.go

```diff
@@ -167,27 +168,53 @@ func resourceVcdDNATCreate(d *schema.ResourceData, meta interface{}) error {
 func resourceVcdDNATRead(d *schema.ResourceData, meta interface{}) error {
 	vcdClient := meta.(*VCDClient)
 
-	e, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
 
 	if err != nil {
 		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
 	}
 
+	// Terraform refresh won't work if Rule was edit in UI. vCD API uses tag elements to map edge gtw Id's
```

Comment:
If I understand correctly if this is lost we will actually mark the resource as deleted - `d.SetId("")`. I think it's important to note that (especially in website documentation)

Reply:
Well in reality this though we use d.SetId("") but return err, so resource won't be removed. This kind'a global issue in our codebase. We do it such behaviour randomly. We aren't consistent. @lvirbalas @dataclouder Seems we need to decided if we don't find something - do throw error and leave user to handle it or do we remove automatically and happy with it.

Reply:
Yes. I know we usually do this, but this is a bit different because user action can make the rule `invisible` to provider and then we will automatically remove it.

Reply:
So options are: throw err as we do know and leave state as it is.
Or Print message and remove resource.

Reply:
Need more info - pros/cons of one approach vs. the other.

Reply:
My initial point was to simply flag it for user (note at the top of resource page) that it can happen so. 


> I think it's important to note that (especially in website documentation)

Reply:
So:
if leave it - we leave user handle state manually. If delete it - user can't do anything if he may fix issue in env he has to add resource back manually or with import when it exist.

Reply:
changed to remove from state